File: Greedy_Algorithms/first_fit_decreasing.py
def sorting(arr,n):
    for i in range(0,n):
        for j in range (0,n-i-1):
            if arr[j]<arr[j+1]:
                arr[j],arr[j+1]=arr[j+1],arr[j]


def firstfit_decreasing(weight,n,c):
    res=0

    bin_rem=[0]*n
    for i in range (n):
        j=0
        while(j<res):
            if (bin_rem[j]>=weight[i]):
                bin_rem[j]=bin_rem[j]-weight[i]
                break
            j+=1

        if (j==res):
            bin_rem[res]=c-weight[i]
            res=res+1
    return res
weight=[2,5,4,7,1,3,8]
c=10
n=len(weight)
sorting(weight,n)
print(weight)
print(firstfit_decreasing(weight,n,c))
            
            

File: Greedy_Algorithms/first_FIt_decreasing.cpp
#include<bits/stdc++.h>
using namespace std;
int ffdecreasing(int weight[],int n,int c){
    int res=0;
    int binrem[n];
    sort(weight,weight+n,greater<int>());

    for (int i=0;i<n;i++){
        int j;
        for (j=0;j<res;j++){
            if (binrem[j]>=weight[i]){
                binrem[j]=binrem[j]-weight[i];
                break;
            }
        }
        if (j==res){
            binrem[res]=c-weight[i];
            res++;

        }
        
    }
    return res;
}
int main(){
    int weight[]={2,5,4,7,1,3,8};
    int n=sizeof(weight)/sizeof(int);
    int c=10;
    cout<<ffdecreasing(weight,n,c);


    return 0;
}